gh-108901: Add `Signature.from_code` method (#108902)
python/cpython: Doc/library/inspect.rst

       .. versionadded:: 3.10
          ``globalns`` and ``localns`` parameters.

   .. classmethod:: Signature.from_code(co)

       Return a :class:`Signature` (or its subclass) object
       for a given :class:`code object <types.CodeType>` ``co``.

       Since code objects do not know anything
       about default values or annotations,

>        they will always be ommited from the signature.

Reviewer: You can suggest using Signature.from_callable() when a function is available.

In from_callable() doc, you may mention from_code() but with a quick reminder about limitations (no default values nor annotations). Or maybe don't mention it there :-)

>        they will always be ommited from the signature.

Reviewer 2: Typo:

```suggestion
       they will always be omitted from the signature.
```

       .. versionadded:: 3.10
          ``globalns`` and ``localns`` parameters.

   .. classmethod:: Signature.from_code(co)

       Return a :class:`Signature` (or its subclass) object

>        for a given :class:`code object <types.CodeType>` ``co``.

Reviewer 2: Use italics rather than backticks when referring to parameters

```suggestion
       for a given :class:`code object <types.CodeType>` *co*.
```

>        Since code objects do not know anything
>        about default values or annotations,
>        they will always be omitted from the signature.

Reviewer 2: Lead with the important thing you're trying to tell the reader, then follow up with the explanation of why it's the case:

```suggestion
       Default values and annotations
       will not be included in the signature,
       since code objects have no knowledge of such things.
```

       It is recommended to use :meth:`Signature.from_callable`

>        when function object is available, it does not have these limitations.

Reviewer: Thanks, it's more explicit and propose a concrete action to avoid these limitations!

>        when function object is available, it does not have these limitations.

Reviewer 2: grammar fix:

```suggestion
       when a function object is available,
       as ``Signature.from_callable()`` does not have the same limitations.
```
